### PR TITLE
feat(scripts): read the Node range from cloudlands-fe engines.node

### PR DESCRIPTION
**Pull request by panghy**
Makes scripts/bootstrap-dev-host.sh derive its supported Node range from packages/cloudlands-fe/package.json engines.node (clauses ^X.Y.Z / >=X, || separated), falling back to the built-in range (^22.22.2 || ^24.15.0 || >=26) when the submodule is not initialized or the field is absent/unparseable. Companion to intent-hq/cloudlands-fe#2423, which adds engines.node plus a gate preflight so fe gates and pnpm install fail fast on an unsupported Node instead of reporting a partial vitest pass (finding from the intent-hq/intent#4835 workspace; root cause intent-hq/intent#4669).

Verification: bash scripts/bootstrap-dev-host.test.sh passes with new fixture/fallback/boundary cases; make doctor prints the derived range; shellcheck clean. No submodule gitlink change.

### CODE REVIEW

File: scripts/bootstrap-dev-host.sh [hunk 1]
```diff
@@ -85,13 +89,44 @@ load_versions() {
...
+  set_node_range "$node_range" || set_node_range "$NODE_RANGE_DEFAULT"
+}
+
+# node_range_floors <range>: prints one "<op> <version>" line per ||-separated
+# clause of an engines.node range, or fails when the range is empty or a clause
+# is neither ^X.Y.Z (same major, at least that release) nor >=X[.Y[.Z]].
+node_range_floors() {
+  local clause count=0 re='^(\^|>=)([0-9]+(\.[0-9]+){0,2})$'
+  local -a clauses
+  IFS='|' read -r -a clauses <<<"${1//[[:space:]]/}"
```
[the remark this answers is not in the document]
Comment (panghy) [reply scripts/bootstrap-dev-host.sh]: Fixed in ff8ecf1: node_range_floors now splits on the two-character || only, and any empty clause (leading/trailing/doubled separator) or stray single | fails the clause regex and takes the built-in-range fallback; tests cover ^20.19.0|>=23.1, "^22.22.2 ||", and "^22.22.2 || || >=26".

[the remark this answers is not in the document]
Comment (panghy) [reply scripts/bootstrap-dev-host.sh]: Fixed in ff8ecf1: node_range_floors now splits on the two-character || only, and any empty clause (leading/trailing/doubled separator) or stray single | fails the clause regex and takes the built-in-range fallback; tests cover ^20.19.0|>=23.1, "^22.22.2 ||", and "^22.22.2 || || >=26".
